app/uk/gov/hmrc/helptosave/actors/UCThresholdManager.scala

```diff
@@ -0,0 +1,192 @@
+
+  import context.dispatcher
+  import ThresholdConnectorProxy.{GetThresholdValue ⇒ GetDESThresholdValue, GetThresholdValueResponse ⇒ GetDESThresholdValueResponse}
+  import ThresholdMongoProxy.{GetThresholdValue ⇒ GetMongoThresholdValue, GetThresholdValueResponse ⇒ GetMongoThresholdValueResponse}
+  import ThresholdMongoProxy.{StoreThresholdValue ⇒ StoreMongoThresholdValue, StoreThresholdValueResponse ⇒ StoreMongoThresholdValueResponse}
+
+  val thresholdConfig = config.get[Config]("uc-threshold").value
+
+  implicit val timeout: Timeout = Timeout(thresholdConfig.get[FiniteDuration]("ask-timeout").value)
+
+  val minBackoff: FiniteDuration = thresholdConfig.get[FiniteDuration]("min-backoff").value
+  val maxBackoff: FiniteDuration = thresholdConfig.get[FiniteDuration]("max-backoff").value
+  val numberOfRetriesUntilWaitDoubles: Int = thresholdConfig.get[Int]("number-of-retries-until-initial-wait-doubles").value
+
+  val storeMongoRetries = exponentialBackoffRetry(
+    minBackoff,
+    maxBackoff,
+    numberOfRetriesUntilWaitDoubles,
+    self,
+    { StoreMongoThresholdValue.apply },
+    scheduler
+  )
+
+  val getDESRetries =
+    exponentialBackoffRetry(
+      minBackoff,
+      maxBackoff,
+      numberOfRetriesUntilWaitDoubles,
+      self,
+      { _: Unit ⇒ GetDESThresholdValue },
+      scheduler
+    )
+
+  def getValueFromDES(): Future[GetDESThresholdValueResponse] =
+    (thresholdConnectorProxy ? GetDESThresholdValue)
+      .mapTo[GetDESThresholdValueResponse]
+
+  def storeValue(amount: Double): Future[StoreMongoThresholdValueResponse] =
+    (thresholdMongoProxy ? StoreMongoThresholdValue(amount))
+      .mapTo[StoreMongoThresholdValueResponse]
+
+  def getValueFromMongo(): Future[GetMongoThresholdValueResponse] =
+    (thresholdMongoProxy ? GetMongoThresholdValue)
+      .mapTo[GetMongoThresholdValueResponse]
+
+  override def receive: Receive = notReady
+
+  def notReady: Receive = {
+
+    case GetDESThresholdValue ⇒
+      logger.info("Trying to get threshold from DES")
+      getValueFromDES() pipeTo self
+
+    case GetDESThresholdValueResponse(result) ⇒
+      handleDESThresholdValue(result)
+
+    case GetMongoThresholdValue ⇒
+      logger.info("Trying to get threshold from mongo")
+      getValueFromMongo() pipeTo self
+
+    case GetMongoThresholdValueResponse(result) ⇒
+      handleMongoThresholdValue(result)
+
+    case GetThresholdValue ⇒
+      stash()
+  }
+
+  def ready(thresholdValue: Double): Receive = {
+
+    case GetThresholdValue ⇒
+      sender() ! GetThresholdValueResponse(thresholdValue)
+
+    case _: StoreMongoThresholdValue ⇒
+      // write value to mongo
+      logger.info("Trying to store threshold value in mongo")
+      storeValue(thresholdValue) pipeTo self
+
+    case StoreMongoThresholdValueResponse(result) ⇒
+      result.fold({
+        e ⇒
+          storeMongoRetries.retry(thresholdValue).fold(
```

Comment:
@AndyHWChung  i would say we are overkilling here - we dont need to retry storing in mongo - even if the call fails we just ignore it as we will anyway do another update after 24h